src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -24,7 +24,7 @@ public static async Task RetrieveClaimsAsync(LdapSettings settings, ClaimsIdenti
             settings.ClaimsCache = new MemoryCache(new MemoryCacheOptions { SizeLimit = settings.ClaimsCacheSize });
         }
 
-        if (settings.ClaimsCache.TryGetValue<IEnumerable<string>>(user, out var cachedClaims))
+        if (settings.ClaimsCache.TryGetValue<IEnumerable<string>>(user, out var cachedClaims) && cachedClaims is not null)
```

Comment:
FYI @Tratcher

Reply:
The suggestion @eerhardt had was to use the forgiveness operator, but that just as iffy.